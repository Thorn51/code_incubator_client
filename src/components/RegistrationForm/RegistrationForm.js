import React from "react";
import "./RegistrationForm.css";

export default class RegistrationForm extends React.Component {
  render() {
    return (
      <form className="registration_form">
        <h2 className="section_title">Register</h2>
        <hr />
        <label htmlFor="registration_first_name">First Name:</label>
        <input
          type="text"
          name="first_name"
          id="registration_first_name"
          placeholder="Enter First Name"
          required
        />
        <label htmlFor="registration_last_name">Last Name:</label>
        <input
          type="text"
          name="last_name"
          id="registration_last_name"
          placeholder="Enter Last Name"
          required
        />
        <label htmlFor="registration_email">Email:</label>
        <input
          type="text"
          placeholder="Enter Email"
          name="email"
          id="registration_email"
          required
        />
        <label htmlFor="registration_password">Password:</label>
        <input
          type="password"
          placeholder="Enter Password"
          name="password"
          id="registration_password"
          required
        />
        <label htmlFor="registration_repeat_password">Repeat Password:</label>
        <input
          type="password"
          name="repeat_password"
          id="registration_repeat_password"
          placeholder="Repeat Password"
          required
        />
        <button type="submit" className="form_button">
          Register
        </button>
      </form>
    );
  }
}
