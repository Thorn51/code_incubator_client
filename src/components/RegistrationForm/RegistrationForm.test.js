import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import RegistrationForm from "./RegistrationForm";

describe("<RegistrationForm />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ReigstrationForm />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders as expected", () => {
    const tree = renderer.create(<RegistrationForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
