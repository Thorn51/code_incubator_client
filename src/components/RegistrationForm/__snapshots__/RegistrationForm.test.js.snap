// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationForm /> renders as expected 1`] = `
<form
  className="registration_form"
  onSubmit={[Function]}
>
  <h2
    className="section_title"
  >
    Register
  </h2>
  <hr />
  <label
    htmlFor="registration_first_name"
  >
    First Name:
  </label>
  <input
    id="registration_first_name"
    name="first_name"
    onChange={[Function]}
    placeholder="Enter First Name"
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="registration_last_name"
  >
    Last Name:
  </label>
  <input
    id="registration_last_name"
    name="last_name"
    onChange={[Function]}
    placeholder="Enter Last Name"
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="nickname"
  >
    Nickname:
  </label>
  <input
    id="registration_nickname"
    name="nickname"
    onChange={[Function]}
    placeholder="Enter Nickname"
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="registration_email"
  >
    Email:
  </label>
  <input
    id="registration_email"
    name="email"
    onChange={[Function]}
    placeholder="Enter Email"
    required={true}
    type="text"
    value=""
  />
  <label
    htmlFor="registration_password"
  >
    Password:
  </label>
  <input
    id="registration_password"
    name="password"
    onChange={[Function]}
    placeholder="Enter Password"
    required={true}
    type="password"
    value=""
  />
  <label
    htmlFor="registration_repeat_password"
  >
    Repeat Password:
  </label>
  <input
    id="registration_repeat_password"
    name="repeat_password"
    onChange={[Function]}
    placeholder="Repeat Password"
    required={true}
    type="password"
    value=""
  />
  <button
    className="form_button"
    disabled="First name is required."
    type="submit"
  >
    Register
  </button>
</form>
`;
